src/test/java/org/json/junit/JSONObjectTest.java

```diff
@@ -2036,8 +2037,8 @@ public void valueToString() {
         map.put("key1", "val1");
         map.put("key2", "val2");
         map.put("key3", "val3");
-        assertTrue("map valueToString() incorrect",
-                jsonObject.toString().equals(JSONObject.valueToString(map))); 
+        JSONAssert.assertEquals("map valueToString() incorrect",
```

Comment:
Please replace this line with code that converts the strings to JSONObjects and compare them using JSONObject.similar()